rust/protocol/src/state/session.rs

```diff
@@ -390,8 +392,9 @@ impl SessionState {
         signed_pre_key_id: SignedPreKeyId,
         base_key: &PublicKey,
     ) -> Result<()> {
+        let signed_pre_key_id: u32 = signed_pre_key_id.into();
         let pending = session_structure::PendingPreKey {
-            pre_key_id: pre_key_id.unwrap_or(0),
+            pre_key_id: pre_key_id.unwrap_or(0.into()).into(),
```

Comment:
Maybe `pre_key_id.map(|id| id.into()).unwrap_or(0)`? I know it's both equivalent and a little longer, but it does make it clear that "0" is a placeholder rather than a real pre-key with ID #0.

Reply:
That makes much more sense!

Reply:
Looks like this missed the rebase.

Reply:
Still shows up in at least one other spot!